CRM/ACL/BAO/ACLEntityRole.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ *  Access Control AclRole.
+ */
+class CRM_ACL_BAO_ACLEntityRole extends CRM_ACL_DAO_ACLEntityRole {
+  public static $_entityTable = NULL;
+
+  /**
+   * Whitelist of possible values for the entity_table field
+   *
+   * @return array
+   */
+  public static function entityTables(): array {
+    if (!self::$_entityTable) {
+      self::$_entityTable = [
+        'civicrm_contact' => ts('Contact'),
+        'civicrm_group' => ts('Group'),
+      ];
+    }
+    return self::$_entityTable;
+  }
+
+  /**
+   * @param array $params
+   *
+   * @return CRM_ACL_DAO_EntityRole
+   */
+  public static function create(&$params) {
+    $dao = new CRM_ACL_DAO_EntityRole();
+    $dao->copyValues($params);
+    $dao->save();
+    return $dao;
```

Comment:
Yikes, this function doesn't call hooks.
Doesn't have to be part of this PR but at some point we need to fix it by replacing it all with `writeRecord`
```suggestion
    return self::writeRecord($params);
```

Reply:
agree, will create a followup PR